api/src/main/java/org/openmrs/module/auditlogweb/api/dao/AuditDao.java

```diff
@@ -373,4 +338,111 @@ private List<Class<?>> getNonAbstractAuditedClasses() {
                 .filter(clazz -> clazz != null && !Modifier.isAbstract(clazz.getModifiers()))
                 .collect(Collectors.toList());
     }
-}
\ No newline at end of file
+
+    // NEW overload for count with entityType
+    public long countRevisionsAcrossEntities(Integer userId, Date startDate, Date endDate, String entityType) {
+        List<Class<?>> classes = getNonAbstractAuditedClasses().stream()
+                .filter(c -> entityType == null || entityType.isEmpty() || c.getSimpleName().equalsIgnoreCase(entityType))
+                .collect(Collectors.toList());
+        return countAcrossEntities(classes, userId, startDate, endDate);
+    }
+
+    private List<AuditEntity<?>> fetchAcrossEntities(List<Class<?>> classes, Integer userId, Date startDate, Date endDate, String sortOrder, int page, int size) {
+        List<AuditEntity<?>> combined = new ArrayList<>();
+        for (Class<?> clazz : classes) {
+            try {
+                List<? extends AuditEntity<?>> revisions = getRevisionsWithFilters(
+                        clazz, page, size, userId, startDate, endDate, sortOrder);
+                combined.addAll(revisions);
+            } catch (Exception ex) {
+                if (isMissingAuditTableException(ex)) {
+                    log.warn("Skipping class {} due to missing audit table or SQL error: {}", clazz.getName(), ex.getMessage());
+                } else {
+                    log.error("Unexpected error while fetching audit logs for class {}: {}", clazz.getName(), ex.getMessage(), ex);
+                }
+            }
+        }
+        return combined;
+    }
```

Comment:
There is a slight issue in here. This fetches `page * size` records from every entity type, combines them all in memory, sorts, then paginates. With many entity types, this could load thousands of records into memory. Can we put a comment here explaing this so we can find a better alternative in future?